src/Validator/AdherentForCommitteeMandateReplacementValidator.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace App\Validator;
+
+use App\Committee\CommitteeAdherentMandateManager;
+use App\Committee\CommitteeMandateCommand;
+use App\Committee\Exception\CommitteeAdherentMandateException;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+class AdherentForCommitteeMandateReplacementValidator extends ConstraintValidator
+{
+    private $mandateManager;
+
+    public function __construct(CommitteeAdherentMandateManager $mandateManager)
+    {
+        $this->mandateManager = $mandateManager;
+    }
+
+    public function validate($value, Constraint $constraint)
+    {
+        if (!$constraint instanceof AdherentForCommitteeMandateReplacement) {
+            throw new UnexpectedTypeException($constraint, CommitteeProvisionalSupervisor::class);
+        }
+
+        if (!$value) {
+            return;
+        }
+
+        if (!$value instanceof CommitteeMandateCommand) {
+            throw new UnexpectedTypeException($value, CommitteeMandateCommand::class);
```

Comment:
could you use `UnexpectedValueException` instead, please. `UnexpectedTypeException` it's for `$constraint` check